### PR TITLE
Chore/rework fullscreen configuration

### LINKED ISSUE #4055: Feature: Hiding Nav Bar
### Version

6.4.3

### What platforms are you having the problem on?

Android

### System Version

Android

### On what device are you experiencing the issue?

Real device, Simulator

### Architecture

Old architecture

### What happened?

When the full-screen view is activated, the default controls are shown, and the mobile bottom navigation bar remains visible, even in full-screen mode. To address this issue, consider upgrading the FullScreenPlayerView found under com => brentvatne => exoplayer.

Code Implementation
To enable immersive mode and hide the navigation bar, override the onAttachedToWindow method:

override fun onAttachedToWindow() {
    super.onAttachedToWindow()
    // Enable immersive mode to hide the navigation bar
    window.decorView.systemUiVisibility = (
        View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
        or View.SYSTEM_UI_FLAG_FULLSCREEN
        or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
        or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
        or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
        or View.SYSTEM_UI_FLAG_LAYOUT_STABLE
    )

    if (reactExoplayerView.preventsDisplaySleepDuringVideoPlayback) {
        mKeepScreenOnHandler.post(mKeepScreenOnUpdater)
    }
}

Resetting UI Visibility
To reset the UI visibility to default, override the onDetachedFromWindow method:

override fun onDetachedFromWindow() {
    super.onDetachedFromWindow()
    // Reset UI visibility to default
    window.decorView.systemUiVisibility = (
        View.SYSTEM_UI_FLAG_LAYOUT_STABLE
        or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
        or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
    )
}

### Reproduction

repository link

### Reproduction

Step to reproduce this bug are: When the full-screen view is activated

### PR DESCRIPTION
**Pull request by freeboub**
## Summary
Handle the status of the navigation bar in full-screen mode.

### Motivation
Fix #4055
maybe replace: https://github.com/TheWidlarzGroup/react-native-video/pull/4067

### Changes
Add a new property to `controlsStyle` called `hideNavigationBarOnFullScreenMode`

## Test plan
If you set this property to `true`, the navigation bar will be hidden; if you set it to `false`, it will be visible.

### CODE REVIEW

**Comment by freeboub** [comment]
@seyedmostafahasani I rework your PR to be more flexible, I also add notification bar visibility in the configuration.
I add some code to be able to test in the sample also.
Please let me know what you think of this version.

Notice That I am not able to find your fork in github UI ... I wanted to point on your PR, but I cannot ...

File: android/src/main/java/com/brentvatne/exoplayer/FullScreenPlayerView.kt [hunk 1]
```diff
@@ -127,4 +163,48 @@ class FullScreenPlayerView(
         layoutParams.setMargins(0, 0, 0, 0)
         return layoutParams
     }
+
+    // Move the UI to fullscreen.
+    // if you change this code, remember to check that the UI is well restored in restoreUIState
+    private fun updateNavigationBarVisibility(
```
Comment (seyedmostafahasani) [inline android/src/main/java/com/brentvatne/exoplayer/FullScreenPlayerView.kt:186]: The method `updateNavigationBarVisibility` can be simplified by extracting the repeated logic for showing or hiding UI components.

```
private fun updateBarVisibility(
    inset: WindowInsetsControllerCompat,
    type: Int,
    shouldHide: Boolean?,
    initialVisibility: Boolean?,
    systemBarsBehavior: Int? = null
) {
    shouldHide?.takeIf { it != initialVisibility }?.let {
        if (it) {
            inset.hide(type)
            systemBarsBehavior?.let { behavior -> inset.systemBarsBehavior = behavior }
        } else {
            inset.show(type)
        }
    }
}

private fun updateNavigationBarVisibility(
    window: Window,
    hideNavigationBarOnFullScreenMode: Boolean?,
    hideNotificationBarOnFullScreenMode: Boolean?,
    systemBarsBehavior: Int?
) {
    val inset = WindowInsetsControllerCompat(window, window.decorView)

    // Update navigation bar visibility and apply systemBarsBehavior if hiding
    updateBarVisibility(
        inset, 
        WindowInsetsCompat.Type.navigationBars(), 
        hideNavigationBarOnFullScreenMode, 
        initialNavigationBarIsVisible, 
        systemBarsBehavior
    )

    // Update notification bar visibility (no need for systemBarsBehavior here)
    updateBarVisibility(
        inset, 
        WindowInsetsCompat.Type.statusBars(), 
        hideNotificationBarOnFullScreenMode, 
        initialNotificationBarIsVisible
    )
}

```
WDYT?

**Comment by seyedmostafahasani** [comment]
@freeboub 
No problem at all! The important thing is improving the library, whether it’s done by you, me, or other collaborators and contributors. I checked your PR, and everything looks good, just I left a small comment :) . Thanks for making those updates!

**Comment by seyedmostafahasani** [comment]
I closed my PR.

File: android/src/main/java/com/brentvatne/exoplayer/FullScreenPlayerView.kt (hunk 1 above)
Comment (freeboub) [reply android/src/main/java/com/brentvatne/exoplayer/FullScreenPlayerView.kt:186]: I agree, I just apply your suggestion, Thank you !

**Review by seyedmostafahasani — APPROVED** [review]

**Comment by freeboub** [comment]
Thank you @seyedmostafahasani , so let's merge this PR

**Pull request merged**